From e06329b6235fcb0791a15ef19bf25d27bc32242b Mon Sep 17 00:00:00 2001
From: Xiang Wang <xiang.a.wang@intel.com>
Date: Fri, 8 May 2015 14:29:45 +0800
Subject: [PATCH] PCI: fix KW issue in pci_disbale_dev_pme_poll

1. Non-void function does not return value
for pci_disbale_dev_pme_poll()
2. correct typo of function name

Change-Id: Id7026f71cef376802d6186460aa80194545d5242
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-35612
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1581
Signed-off-by: Xiang Wang <xiang.a.wang@intel.com>
---
 drivers/pci/quirks.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 76ef064..b89ba6b 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -2920,9 +2920,10 @@ static void quirk_pcie_enable_rtpm(struct pci_dev *dev)
 DECLARE_PCI_FIXUP_ENABLE(PCI_VENDOR_ID_INTEL,
 	PCI_DEVICE_ID_INTEL_CHV_PCIe_0, quirk_pcie_enable_rtpm);
 
-static int pci_disbale_dev_pme_poll(struct pci_dev *pdev, void *data)
+static int pci_disable_dev_pme_poll(struct pci_dev *pdev, void *data)
 {
 	pdev->pme_poll = false;
+	return 0;
 }
 
 /*PCIEe port 1 on Cherryview should support runtime PM and ignore children*/
@@ -2937,7 +2938,7 @@ static void quirk_pcie_enable_rtpm_ignore_children(struct pci_dev *dev)
 	 * no matter the status of child device.
 	 * Set device pme_poll as false.
 	 */
-	pci_walk_bus(dev->subordinate, pci_disbale_dev_pme_poll, NULL);
+	pci_walk_bus(dev->subordinate, pci_disable_dev_pme_poll, NULL);
 	pm_runtime_put_noidle(&dev->dev);
 	pm_runtime_allow(&dev->dev);
 }
-- 
1.9.1

